src/View/AbstractDirectoryView.vala

```diff
@@ -2103,26 +2102,27 @@ namespace FM {
 
                 filter_this_app_from_open_with_apps ();
 
-                if (open_with_apps != null) {
+                if (open_with_apps.length () > 0) {
                     var apps_section = new GLib.Menu ();
                     int index = -1;
                     int count = 0;
                     string last_label = "";
                     string last_exec = "";
 
-                    foreach (var app in open_with_apps) {
+                    foreach (var app_info in open_with_apps) {
```

Comment:
GLib.List allows unowned iteration, better use it to not add reference count without any reason ;)
```suggestion
                    foreach (unowned GLib.AppInfo app_info in open_with_apps) {
```